app/controllers/dois_controller.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+class DoisController < ApplicationController
+  class DoiIsMintingError < RuntimeError; end
+
+  def create
+    resource = find_resource
+
+    authorize(resource, :edit?)
+    raise DoiIsMintingError if doi_present_or_minting?(resource)
+
+    MintDoiAsync.call(resource)
+  rescue DoiIsMintingError
+    # I don't think we need to do anything special here, just skip over the call
+    # to MintDoiAsync and redirect back where we came from?
+  ensure
+    redirect_to resource_path(params[:resource_id])
+  end
+
+  def show
+    @resource = find_resource
+    render :show, layout: false
```

Comment:
@rschenk I'm also seeing this not getting covered either, but you're redirecting to it in the tests.

Reply:
Come to think of it, this controller action is not currently used at all. I intended for it to feed an ajax auto-refresher when the DOI is minting, but I never built that auto-refresher. I think I'll remove it and its associated view for now. All it did was render a MintingStatusDoiComponent with no layout

Reply:
Yay for less code!